primitives/xcm/src/barriers.rs

```diff
@@ -35,9 +35,9 @@ use xcm_executor::traits::ShouldExecute;
 /// first
 pub struct AllowTopLevelPaidExecutionDescendOriginFirst<T>(PhantomData<T>);
 impl<T: Contains<MultiLocation>> ShouldExecute for AllowTopLevelPaidExecutionDescendOriginFirst<T> {
-	fn should_execute<Call>(
+	fn should_execute<RuntimeCall>(
```

Comment:
it's named `Call{ instead of `RuntimeCall` on purpose, tha transact call type is not necessarily the same type as the runtime call.